internal/api/rbac/policy_rules.go

```diff
@@ -150,3 +151,57 @@ func buildRule(
 	}
 	return rule
 }
+
+// nolint: goconst
+func normalizeResourceTypeName(resource string) (string, error) {
```

Comment:
Postel's rule. Map incorrect, singular names to correct plural names. Strip whitespace. Ensure lower case.

Reply:
The one out of this list that gives me pause is: 
> Map incorrect, singular names to correct plural names

Instead of normalizing the plurality, I think I would prefer rejecting the non-plural or unsupported resources (with list of accepted ones). This goes back to being equal to kubernetes rbac rules. With this in place, a user might think they are doing the right thing (because when they apply it via API it works), but the moment they try the same rule with k8s, it would fail.

Reply:
Does it make any difference to know that `kargo get role <name> --as-kubernetes-resources` exists?

It's specifically for the eventuality of making the leap from managing roles imperatively to gitopsing them.

Reply:
Oh... also... I think they _can't_ get it wrong via UI and via CLI, users do things like this:

```shell
kargo grant --role <name> --resource-type <type> --verb get --verb list
```

This is where I was most expecting that incorrect singular variants of resource type names would occur.

Reply:
> but the moment they try the same rule with k8s, it would fail.

Actually that would remain true with whitespace, and non-lowercase... So perhaps I am suggesting we be strict about this.

Reply:
We hammered this out offline and the PR has been amended to validate resource types instead of normalizing them.